pulsar-client/src/main/java/org/apache/pulsar/client/impl/ClientCnx.java

```diff
@@ -747,7 +747,7 @@ protected void handleCloseProducer(CommandCloseProducer closeProducer) {
     protected void handleCloseConsumer(CommandCloseConsumer closeConsumer) {
         log.info("[{}] Broker notification of Closed consumer: {}", remoteAddress, closeConsumer.getConsumerId());
         final long consumerId = closeConsumer.getConsumerId();
-        ConsumerImpl<?> consumer = consumers.get(consumerId);
+        ConsumerImpl<?> consumer = consumers.remove(consumerId);
```

Comment:
Nice catch! Could you please add a test?

Reply:
Done.